VAMobile/src/screens/HealthScreen/Appointments/AppointmentTypeComponents/SharedComponents/AppointmentDetailsModality.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+
+import { TFunction } from 'i18next'
+
+import { AppointmentAttributes, AppointmentStatusDetailTypeConsts } from 'api/types'
+import { Box, TextView } from 'components'
+import { NAMESPACE } from 'constants/namespaces'
+import { a11yLabelVA } from 'utils/a11yLabel'
+import {
+  AppointmentDetailsScreenType,
+  AppointmentDetailsSubType,
+  AppointmentDetailsSubTypeConstants,
+  AppointmentDetailsTypeConstants,
+} from 'utils/appointments'
+import { useTheme } from 'utils/hooks'
+
+type AppointmentDetailsModalityProps = {
+  attributes: AppointmentAttributes
+  type: AppointmentDetailsScreenType
+  subType: AppointmentDetailsSubType
+}
+
+const modalityHeader = (subType: AppointmentDetailsSubType, type: AppointmentDetailsScreenType, t: TFunction) => {
+  let appointmentHeaderType = ''
+  switch (type) {
+    case AppointmentDetailsTypeConstants.InPersonVA:
+      appointmentHeaderType = t('appointments.inPersonVA.upcomingTitle')
+      break
+    default:
+      appointmentHeaderType = ''
+  }
+
+  switch (subType) {
```

Comment:
it would be possible, but I am going to leave it as two separate functions for readability purposes once we map out all the types and subtypes. Maybe when we get to the last ticket we can re evaluate if this can be combined without sacrificing readability at that point.